File: book-shop-admin/src/app/common/Product.ts
import { Category } from "./Category";

export class Product {
    'productId':number;
    'name':string;
    'quantity': number;
    'price': number;
    'discount':number;
    'image': string;
    'description': string;
    'enteredDate': Date;
    'category': Category;
    'status': boolean;
    'sold': number;
    'author':string;
    'pageSize':number;
    'suitableReadingAge':number;
    'language':string;
    'size':string;
    'publisher':string;

  constructor(id:number, name:string) {
    this.productId = id;
    this.name = name;
  }
}

File: book-shop/src/main/java/com/shop/entitty/model/Product.java
package com.shop.entitty.model;

import java.io.Serializable;
import java.time.LocalDate;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.shop.entitty.base.AbstractAuditingEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@SuppressWarnings("serial")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "products")
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Product extends AbstractAuditingEntity {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long productId;
	private String name;
	private int quantity;
	private Double price;
	private int discount;
	private String image;
	private String description;
	private LocalDate enteredDate;
	private Boolean status;
	private int sold;
	private String author;
	private int pageSize;
	private int suitableReadingAge;
	private String language;
	private String size;
	private String publisher;

	@ManyToOne
	@JoinColumn(name = "categoryId")
	private Category category;
}
